### ISSUE TITLE
jinja[invalid]: Unhandled exception when ipaddr filter used in vars

### ISSUE THREAD

**Issue by bluikko** [ansible/ansible-lint#2811]
##### Summary

If the `ansible.utils.ipaddr` filter is used from `vars:` then a false positive `jinja[invalid]` is raised due to `An unhandled exception`. The role works when run in `ansible-playbook` and also functionally exactly equivalent expression works if the expression is written without using `vars:`.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.9.1 using ansible 2.14.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

The problematic task can be simplified to reduced form:
```console (paste below)
- name: Test if address is in a subnet  # <--- works
  ansible.builtin.set_fact:
    site_address: "{{ var_address | ansible.utils.ipaddr('192.0.2.0/24') }}"
- name: Test if address is in a subnet using vars  # <--- false positive
  ansible.builtin.set_fact:
    site_address: "{{ filtered_ips }}"
  vars:
    filtered_ips: "{{ var_address | ansible.utils.ipaddr('192.0.2.0/24') }}"
```

Since the real play is much more complex I tried relying on `var_address` as an `extra_vars` in ansible-lint configuration but it changed nothing.

##### Desired Behavior

The usage of `vars` should not make difference when using the `ipaddr` filter.

##### Actual Behavior

```paste below
jinja[invalid]: An unhandled exception occurred while templating '{{ var_address | ansible.utils.ipaddr('192.0.2.0/24') }}'. Error was a <class 'ansible.errors.AnsibleFilterError'>, original message: Unrecognized type <<class 'ansible.template.AnsibleUndefined'>> for ipaddr filter <value>
roles/server/tasks/main.yml:5 Task/Handler: Make site address that fails when using vars
```

**Comment by bluikko** [comment]
This has surfaced again after testing update from 6.11.x to 6.14.0 but the error looks a bit different and completely different task. All cases of `jinja[invalid]` seem to relate to `ipaddr` when using a var that has been `lookup()`ed or otherwise coming from external source (like an extra var). Trying to narrow it down to a test case...
```
jinja[invalid]: Unrecognized type <<class 'ansible.template.AnsibleUndefined'>> for ipaddr filter <value>
```
